src/Reflex/Dom/Contrib/Widgets/ScriptDependent.hs

```diff
@@ -0,0 +1,140 @@
+{-# LANGUAGE FlexibleContexts    #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE JavaScriptFFI       #-}
+{-# LANGUAGE LambdaCase          #-}
+-- | This module provides a method of expressing widgets that depend on an external javascript source.
+module Reflex.Dom.Contrib.Widgets.ScriptDependent
+  ( widgetHoldUntilDefined
+  ) where
+
+import           Control.Concurrent          (forkIO, threadDelay)
+import           Control.Monad               (void)
+import           Control.Monad.IO.Class      (MonadIO, liftIO)
+import           Reflex.Dom
+#ifdef ghcjs_HOST_OS
+import           Control.Concurrent.STM.TVar
+import           Control.Monad.STM (atomically)
+import           Data.Function               (fix)
+import           Data.Time.Clock             (NominalDiffTime, diffUTCTime, getCurrentTime)
+import           GHCJS.Foreign.Callback
+import           Data.JSString (JSString, pack)
+#else
+import           Data.Time.Clock             (NominalDiffTime)
+#endif
+
+
+#ifdef ghcjs_HOST_OS
+
+------------------------------------------------------------------------------
+-- | A 15 second timeout.
+timeout :: NominalDiffTime
+timeout = 15.0
+
+
+------------------------------------------------------------------------------
+-- | Determines if a symbol is defined on the window.
+foreign import javascript unsafe
+ "(function () { \
+   return (typeof window[$1] != 'undefined'); \
+  })()"
+  definedInWindow_js :: JSString -> IO Bool
+
+
+------------------------------------------------------------------------------
+-- | Determines if a symbol is defined on the window.
+definedInWindow :: String -> IO Bool
+definedInWindow = definedInWindow_js . pack
+
+
+------------------------------------------------------------------------------
+-- | Does a script tag injection.
+foreign import javascript unsafe
+  "(function () { \
+     var script = document.createElement('script'); \
+         script.type   = 'text/javascript'; \
+         script.async  = true; \
+         script.onload = function () { \
+           console.log('loaded script ' + $1); \
+           $2(); \
+         }; \
+         script.src = $1; \
+     console.log('loading script ' + $1); \
+     document.getElementsByTagName('head')[0].appendChild(script); \
+   })()"
+  injectScript_js :: JSString -> Callback (IO ()) -> IO ()
+
+
+------------------------------------------------------------------------------
+-- | Inject a script into the current document and wait for it to load or timeout.
+-- The timeout is set to 15 seconds.
+injectScript :: String -> IO () -> IO ()
+injectScript url innerCB = do
+  tvar    <- newTVarIO False
+  outerCB <- asyncCallback $ atomically $ writeTVar tvar True
+  start   <- getCurrentTime
+  injectScript_js (pack url) outerCB
+  void $ forkIO $ fix $ \loop -> readTVarIO tvar >>= \case
+    True  -> releaseCallback outerCB >> innerCB
+    False -> do
+      threadDelay 10
+      now <- getCurrentTime
+      if diffUTCTime now start < timeout
```

Comment:
I honestly didn't know about the `timeout` function in base. I'm guessing you're talking about `System.Timeout.timeout` ?